src/plugins/data/server/index_patterns/routes/scripted_fields/put_scripted_field.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { IRouter } from '../../../../../../core/server';
+import { handleErrors } from '../util/handle_errors';
+import { fieldSpecSchema } from '../util/schemas';
+import type { IndexPatternsServiceProvider } from '../../index_patterns_service';
+
+export const registerPutScriptedFieldRoute = (
+  router: IRouter,
+  indexPatternsProvider: IndexPatternsServiceProvider
+) => {
+  router.put(
+    {
+      path: '/api/index_patterns/index_pattern/{id}/scripted_field',
+      validate: {
+        params: schema.object(
+          {
+            id: schema.string({
+              minLength: 1,
+              maxLength: 1_000,
+            }),
+          },
+          { unknowns: 'allow' }
+        ),
+        body: schema.object({
+          refresh_fields: schema.maybe(schema.boolean({ defaultValue: false })),
+          field: fieldSpecSchema,
+        }),
+      },
+    },
+    router.handleLegacyErrors(
+      handleErrors(async (ctx, req, res) => {
+        const savedObjectsClient = ctx.core.savedObjects.client;
+        const elasticsearchClient = ctx.core.elasticsearch.client.asCurrentUser;
+        const indexPatternsService = await indexPatternsProvider.createIndexPatternsService(
+          savedObjectsClient,
+          elasticsearchClient
+        );
+        const id = req.params.id;
+        const {
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          refresh_fields = true,
+          field,
+        } = req.body;
+
+        if (!field.scripted) {
+          throw new Error('Only scripted fields can be put.');
+        }
+
+        const indexPattern = await indexPatternsService.get(id);
+
+        const oldFieldObject = indexPattern.fields.getByName(field.name);
+        if (!!oldFieldObject) {
+          indexPattern.fields.remove(oldFieldObject);
+        }
+
+        indexPattern.fields.add({
+          ...field,
+          aggregatable: true,
+          searchable: true,
+        });
+
+        await indexPatternsService.updateSavedObject(indexPattern);
+        if (refresh_fields) {
```

Comment:
I think this will already have happened in line 68.

Seems like there's a few of these in different files. 

I think running `refreshFields()` was only needed on index pattern creation. It happens automatically part of the `get` call.

Reply:
Removed. I'll remove it in all API endpoints except the index pattern creation one.